jumpscale/tools/notificationsqueue/queue.py

```diff
@@ -0,0 +1,120 @@
+from enum import Enum
+
+from jumpscale.loader import j
+
+
+class LEVEL(Enum):
+    INFO = "info"
+    WARNING = "warning"
+    ERROR = "error"
+
+
+class Notification:
+    def __init__(self):
+        self.id = None
+        self.category = None
+        self.level = 0
+        self.message = None
+        self.date = None
+
+    @classmethod
+    def loads(cls, value):
+        json = j.data.serializers.json.loads(value)
+        instance = cls()
+        instance.__dict__ = json
+        return instance
+
+    @property
+    def json(self):
+        return self.__dict__
+
+    def dumps(self):
+        return j.data.serializers.json.dumps(self.__dict__)
+
+
+class NotificationsQueue:
+    def __init__(self, *args, **kwargs):
+        self._rkey = "queue:notifications"
+        self._rkey_incr = "queue:notifications:id:incr"
+        self._rkey_seen = "list:notifications:seen"
+        self._seen_list_max_size = 10
+        self._db = None
+
+    @property
+    def db(self):
+        if self._db is None:
+            self._db = j.core.db
+        return self._db
+
+    def push(self, message: str, category: str = "default", level: LEVEL = LEVEL.INFO):
+        """Push a new notification
+
+        Arguments:
+            message {str} -- notification message
+
+        Keyword Arguments:
+            type {str} -- notification type (default: {"default"})
+            level {LEVEL} -- notification level (default: {LEVEL.INFO})
+        """
+        if not self.db.is_running():
+            return
+
+        if not isinstance(level, LEVEL):
+            raise j.exceptions.Value(f"{level} is not of type: LEVEL")
+
+        notification = Notification()
+
+        notification.message = message
+        notification.level = level.value
+        notification.category = category
+        notification.date = j.data.time.utcnow().timestamp
+        notification.id = self.db.incr(self._rkey_incr)
+
+        self.db.lpush(self._rkey, notification.dumps())
+
+    def fetch(self, count: int = -1) -> list:
+        """Fetch notifications from queue
+
+        Keyword Arguments:
+            count {int} -- number of new notifications to fetch (default: {-1} = new notifications only)
+                           if new notifications < count => will return list of (new notifications + old notifications) with length = count
+
+        Returns:
+            list on Notification objects
```

Comment:
Also the style here, might be:

```suggestion
            list: notification objects
```